modules/data-entry/src/main/java/ca/sickkids/ccm/lfs/FilterServlet.java

```diff
@@ -116,65 +116,89 @@ public void doGet(final SlingHttpServletRequest request, final SlingHttpServletR
      * Copies over lfs:Question fields from the input JsonObject, optionally handling questions that
      * may already exist in the builder.
      *
-     * @param questions A JsonObject (from an lfs:Questionnaire) whose fields may be lfs:Questions
+     * @param questions A JsonObject (from an lfs:Questionnaire or lfs:Section) whose fields may be lfs:Questions
      * @param builder A JsonObjectBuilder to copy results to
      * @param seenTypes Either a map from field names to dataTypes, or null to disable tracking
      * @param seenElements Either a map from field names to jcr:uuids, or null to disable tracking
      * @return the content matching the query
      */
-    private void copyFields(JsonObject questions, JsonObjectBuilder builder, Map<String, String> seenTypes,
+    private void copyQuestions(JsonObject questions, JsonObjectBuilder builder, Map<String, String> seenTypes,
             Map<String, String> seenElements)
     {
         // Copy over the keys
         for (String key : questions.keySet()) {
             // Skip over non-questions (non-objects)
-            if (questions.get(key).getValueType() != ValueType.OBJECT
-                || !questions.getJsonObject(key).getString("jcr:primaryType").equals("lfs:Question")) {
+            if (questions.get(key).getValueType() != ValueType.OBJECT) {
                 continue;
             }
+            JsonObject datum = questions.getJsonObject(key);
 
-            if (seenTypes == null) {
-                // No map to keep track of dataTypes provided: add blindly to the builder
-                builder.add(key, questions.get(key));
-                continue;
+            // Copy over information from children of sections
+            if (datum.getString("jcr:primaryType").equals("lfs:Section")) {
```

Comment:
```suggestion
            if ("lfs:Section".equals(datum.getString("jcr:primaryType"))) {
```
^ This is safer even though the generated JSON always specifies primary types for objects at this time.